ctd/read.py

```diff
@@ -261,7 +264,8 @@ def from_btl(fname):
 
     df["Statistic"] = df["Statistic"].str.replace(r"\(|\)", "")  # (avg) to avg
 
-    name = _basename(fname)[1]
+    if name is None:
+        name = _basename(fname)[1]
```

Comment:
This Forces folks using stream to always pass a name but that is optional, so the API here is kind of broken. (At least lacking documentation.)

Ideally we should check if it is stream and hardcode a default name for those cases.

For that to work we should move this check to `_read_file` and make it output both `f` and `name` there. That way we check for StringIO only once.

Reply:
I hope I'm not over complicating this idea, but did you want `_read_file` to return a tuple of values, the first being the StringIO object and the second being the file/stream name?

Reply:
Yes. `_read_file` is an internal use function, so it is OK to "break it" and better than changing the call on `from_btl`. Does that make sense?

Reply:
Here's an idea.

What if when reading the file, we grab the BTL "File Name" in the header section and use that to set the file name?

Reply:
Sorry, what I mean is what if we got the file name in the _parse_seabird() method, if a name hasn't been set already.